### PR TITLE
Rpmdb support

### PR DESCRIPTION
**Pull request by cmaritan**
Address #254

### CODE REVIEW

**Review by another-rex — COMMENTED** [review]
Thanks! Looks like a reasonable implementation to me, my main concern is pulling in the relatively large number of dependencies dependencies to parse the sqlite database., especially before any support for redhat advisories are included in osv.dev .

File: pkg/lockfile/rpmdb.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			continue
+		}
+
+		var pkg = PackageDetails{
+			Ecosystem: RedHatEcosystem,
+			CompareAs: RedHatEcosystem,
+		}
+		pkg.Name = rpmPkg.Name
+		pkg.Version = rpmPkg.Version
+		packages = append(packages, pkg)
```
Comment (G-Rath) [inline pkg/lockfile/rpmdb.go:45]: We might as well just do all together:

```suggestion
	packages = append(packages, PackageDetails{
		Name: rpmPkg.Name,
		Version: rpmPkg.Version,
		Ecosystem: RedHatEcosystem,
		CompareAs: RedHatEcosystem,
	})
```

File: pkg/lockfile/rpmdb_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		{
+			Name:      "htop",
+			Version:   "3.2.1",
+			Ecosystem: lockfile.RedHatEcosystem,
+			CompareAs: lockfile.RedHatEcosystem,
+		},
+	})
+}
+
+func TestParseRpmDB_SQLite_FileDoesNotExist(t *testing.T) {
```
Comment (G-Rath) [inline pkg/lockfile/rpmdb_test.go:48]: nit: I typically had these kind of checks at the top as they're usually pretty short and test the first lines of the parser - while by itself not a big deal, I think we should still move these to be consistent with the existing tests

**Review by G-Rath — COMMENTED** [review]
This looks cool! It would be good if you could have a look over the other parsers and consider if there's any situations they account for that should be handled for this parser/"lockfile" too, such as the same packages being listed multiple times

(I'm not super familiar with RedHat stuff, so it might all be being handled by the library)

**Comment by cmaritan** [comment]
> Thanks! Looks like a reasonable implementation to me, my main concern is pulling in the relatively large number of dependencies dependencies to parse the sqlite database., especially before any support for redhat advisories are included in osv.dev .

Thanks for the feedback, I hadn't considered this point.

If it's a problem, we can surely keep aside this PR and merge it when osv.dev will have redhat advisories support.

**Comment by cmaritan** [comment]
> This looks cool! It would be good if you could have a look over the other parsers and consider if there's any situations they account for that should be handled for this parser/"lockfile" too, such as the same packages being listed multiple times
> 
> (I'm not super familiar with RedHat stuff, so it might all be being handled by the library)

Usually rpmdb are written only by librpm, passing duplicates/invalid values should not be possible. 
Generating this test case would require direct insert of records inside sqlite db but maintainers keep its structure [undocumented](https://github.com/rpm-software-management/rpm/discussions/2211#discussioncomment-3793589) so it's possible to break other things.

**Review by Spiritaine — APPROVED** [review]

**Comment by andrewpollock** [comment]
Rocky Linux (currently present in OSV.dev) uses RPM and I'm not sure about Mageia, but it's in progress... It might be worthwhile getting this PR merged?

**Comment by picatz** [comment]
👋 Is there any status update on this PR? Would love to see this feature land.

**Comment by another-rex** [comment]
@picatz Still in progress and is being actively worked on, rough estimate is around the end of October, there should be some significant progress by then.

**Comment by picatz** [comment]
Awesome, I really appreciate the quick update @another-rex!

**Comment by jessesmd** [comment]
Please also consider the epoch related to rpm packages as they have a specific format epoch:name-version-release.arch
Thank you!

**Pull request closed without merging**